Tests iOS/MyList/SearchTests.swift

```diff
@@ -0,0 +1,104 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import XCTest
+import Sails
+
+class SearchTests: XCTestCase {
+    var server: Application!
+    var app: PocketAppElement!
+
+    override func setUpWithError() throws {
+        continueAfterFailure = false
+
+        let uiApp = XCUIApplication()
+        app = PocketAppElement(app: uiApp)
+
+        server = Application()
+
+        server.routes.post("/graphql") { request, _ in
+            let apiRequest = ClientAPIRequest(request)
+
+            if apiRequest.isForSlateLineup {
+                return Response.slateLineup()
+            } else if apiRequest.isForSavesContent {
+                return Response.saves()
+            } else if apiRequest.isForArchivedContent {
+                return Response.archivedContent()
+            } else if apiRequest.isForTags {
+                return Response.emptyTags()
+            } else {
+                fatalError("Unexpected request")
+            }
+        }
+
+        try server.start()
+    }
+
+    override func tearDownWithError() throws {
+        try server.stop()
+        app.terminate()
+    }
+
+    // MARK: - Empty States
+    func test_search_forFreeUser_showsEmptyStateView() {
+        server.routes.post("/graphql") { request, _ in
+            Response.saves("initial-list-free-user")
+        }
+        tapSearch()
+        XCTAssertTrue(app.saves.searchView.exists)
+        XCTAssertTrue(app.saves.searchEmptyStateView(for: "search-empty-state").exists)
+
+    }
+
+    func test_search_forPremiumUser_showsEmptyStateView() {
+        tapSearch()
+        XCTAssertTrue(app.saves.searchView.exists)
+        XCTAssertTrue(app.saves.searchEmptyStateView(for: "recent-search-empty-state").exists)
+    }
```

Comment:
We are testing the different view models via Unit Testing depending on the scope and user type. Do we want to add more cases or those this suffice for now and we can have a more end to end tests as we further implement search?